src/libraries/System.Net.Security/src/System/Net/Security/MD4.cs

```diff
@@ -0,0 +1,255 @@
+            /* Transform as many times as possible. */
+            if (cbSize >= partLen)
+            {
+                Buffer.BlockCopy(array, ibStart, buffer, index, partLen);
+                MD4Transform(state, buffer, 0);
+
+                for (i = partLen; i + 63 < cbSize; i += 64)
+                {
+                    MD4Transform(state, array, ibStart + i);
+                }
+
+                index = 0;
+            }
+
+            /* Buffer remaining input */
+            Buffer.BlockCopy(array, ibStart + i, buffer, index, (cbSize - i));
+        }
+
+        private static byte[] HashFinal()
+        {
+            /* Save number of bits */
+            byte[] bits = new byte[8];
+            Encode(bits, count);
+
+            /* Pad out to 56 mod 64. */
+            uint index = ((count[0] >> 3) & 0x3f);
+            int padLen = (int)((index < 56) ? (56 - index) : (120 - index));
+            HashCore(Padding(padLen), 0, padLen);
+
+            /* Append length (before padding) */
+            HashCore(bits, 0, 8);
+
+            /* Store state in digest */
+            Encode(digest, state);
+
+            return digest;
+        }
+
+        //--- private methods ---------------------------------------------------
+
+        private static byte[] Padding(int nLength)
+        {
+            byte[] padding = new byte[nLength];
+            padding[0] = 0x80;
+            return padding;
+        }
+
+        /* F, G and H are basic MD4 functions. */
+        private static uint F(uint x, uint y, uint z)
+        {
+            return (uint)(((x) & (y)) | ((~x) & (z)));
+        }
+
+        private static uint G(uint x, uint y, uint z)
+        {
+            return (uint)(((x) & (y)) | ((x) & (z)) | ((y) & (z)));
+        }
+
+        private static uint H(uint x, uint y, uint z)
+        {
+            return (uint)((x) ^ (y) ^ (z));
+        }
+
+        /* ROTATE_LEFT rotates x left n bits. */
+        private static uint ROL(uint x, byte n)
+        {
+            return (uint)(((x) << (n)) | ((x) >> (32 - (n))));
+        }
+
+        /* FF, GG and HH are transformations for rounds 1, 2 and 3 */
+        /* Rotation is separate from addition to prevent recomputation */
+        private static void FF(ref uint a, uint b, uint c, uint d, uint x, byte s)
+        {
+            a += F(b, c, d) + x;
+            a = ROL(a, s);
+        }
+
+        private static void GG(ref uint a, uint b, uint c, uint d, uint x, byte s)
+        {
+            a += G(b, c, d) + x + 0x5a827999;
+            a = ROL(a, s);
+        }
+
+        private static void HH(ref uint a, uint b, uint c, uint d, uint x, byte s)
+        {
+            a += H(b, c, d) + x + 0x6ed9eba1;
+            a = ROL(a, s);
+        }
+
+        private static void Encode(byte[] output, uint[] input)
+        {
+            for (int i = 0, j = 0; j < output.Length; i++, j += 4)
+            {
+                if (BitConverter.IsLittleEndian)
```

Comment:
You do need the `BitConverter.IsLittleEndian` check here. `BinaryPrimitives.WriteUInt32LittleEndian` will work correctly on both big and little endian machines.